packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -1090,6 +1204,9 @@ class TransformLayer extends OffsetLayer {
   void applyTransform(Layer child, Matrix4 transform) {
     assert(child != null);
     assert(transform != null);
+    if (_lastEffectiveTransform == null && this.transform != null) {
```

Comment:
What if transform is also null? Wouldn't line 1210 throw then? If it is guaranteed to not be null, maybe add an assert instead?